### PR TITLE
Add feedback modal and product mvp feedback modal components

### PR DESCRIPTION
**Pull request by octaedro**
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:

This PR adds the components FeedbackModal and ProductMVPFeedbackModal to the package `customer-effort-score`.

I think that it would be a good follow-up for this PR to add the storybooks for both components.

- [ ] This PR is a very minor change/addition and does not require testing instructions (if checked you can ignore/remove the next section).

### How to test the changes in this Pull Request:

To test the changes added in this PR you can follow the steps described on [this PR](https://github.com/woocommerce/woocommerce/pull/36544).

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [x] Have you written new tests for your changes, as applicable?
-   [x] Have you created a changelog file for each project being changed, ie `pnpm --filter=<project> changelog add`?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

File: packages/js/customer-effort-score/src/feedback-modal/feedback-modal.tsx [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+ * @param {string}   props.title                Title displayed in the modal.
+ * @param {string}   props.description          Description displayed in the modal.
+ * @param {string}   props.isSendButtonDisabled Boolean to enable/disable the send button.
+ * @param {string}   props.sendButtonLabel      Label for the send button.
+ * @param {string}   props.cancelButtonLabel    Label for the cancel button.
+ * @param {Function} props.onCloseModal         Callback for when user closes modal by clicking cancel.
+ * @param {Function} props.children             Children to be rendered.
+ */
+function FeedbackModal( {
+	onSendFeedback,
```
Comment (joshuatf) [inline packages/js/customer-effort-score/src/feedback-modal/feedback-modal.tsx]: Nitpicky, but I think we could keep callback names generic and not tie it to the component name.

```suggestion
	onSubmit,
```

File: packages/js/customer-effort-score/src/feedback-modal/feedback-modal.tsx [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+	title,
+	description,
+	onCloseModal,
```
Comment (joshuatf) [inline packages/js/customer-effort-score/src/feedback-modal/feedback-modal.tsx]:
```suggestion
	onModalClose,
```

File: packages/js/customer-effort-score/src/feedback-modal/feedback-modal.tsx [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+	children,
+	isSendButtonDisabled,
```
Comment (joshuatf) [inline packages/js/customer-effort-score/src/feedback-modal/feedback-modal.tsx]: Updating the naming here can make this less confusing in the future if we no longer use the word "send."
```suggestion
	isSubmitButtonDisabled,
```

But I wonder if we could shorten this to `canSubmit`?

File: packages/js/customer-effort-score/src/feedback-modal/feedback-modal.tsx [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+	sendButtonLabel,
```
Comment (joshuatf) [inline packages/js/customer-effort-score/src/feedback-modal/feedback-modal.tsx]:
```suggestion
	submitButtonLabel,
```

File: packages/js/customer-effort-score/src/feedback-modal/feedback-modal.tsx [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+		<Modal
+			className="woocommerce-feedback-modal"
+			title={ title }
+			onRequestClose={ closeModal }
+			shouldCloseOnClickOutside={ false }
+		>
+			<Text
+				variant="body"
+				as="p"
+				className="woocommerce-feedback-modal__intro"
```
Comment (joshuatf) [inline packages/js/customer-effort-score/src/feedback-modal/feedback-modal.tsx]:
```suggestion
				className="woocommerce-feedback-modal__description"
```

File: packages/js/customer-effort-score/src/feedback-modal/test/index.tsx [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+import { createElement } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { FeedbackModal } from '../index';
+
+const mockRecordScoreCallback = jest.fn();
+
+describe( 'FeedbackModal', () => {
```
Comment (joshuatf) [inline packages/js/customer-effort-score/src/feedback-modal/test/index.tsx:14]: 🎉  Thanks for these tests!  💯

File: packages/js/customer-effort-score/src/feedback-modal/feedback-modal.tsx [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+			</div>
+		</Modal>
+	);
+}
+
+FeedbackModal.propTypes = {
+	onSendFeedback: PropTypes.func.isRequired,
+	title: PropTypes.string,
+	description: PropTypes.string,
+	defaultScore: PropTypes.number,
```
Comment (joshuatf) [inline packages/js/customer-effort-score/src/feedback-modal/feedback-modal.tsx]: Is this being used?

**Review by joshuatf — COMMENTED** [review]
Looks great, Fernando!  Left a few very nitpicky naming comments and one additional prop I think is no longer being used.

File: packages/js/customer-effort-score/src/feedback-modal/feedback-modal.tsx (hunk 7 above)
Comment (octaedro) [reply packages/js/customer-effort-score/src/feedback-modal/feedback-modal.tsx]: Good eyes!

**Review by joshuatf — APPROVED** [review]
Thanks for the changes!  This LGTM 🚢
